File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.cpp
class Solution {
public:
    static bool comp(vector<int> &a, vector<int> &b){
        if(a[1]==b[1])
            return a[0]<b[0];
        return a[1]<b[1];
    }
    int findLongestChain(vector<vector<int>>& p) {
        int n = p.size(),ans=1;
        sort(p.begin(),p.end(),comp);
        vector<int> prev = p[0];
        for(int i=1;i<n;i++){
            if(p[i][0]>prev[1]){
                ans++;
                prev=p[i];
            }
        }
        return ans;
    }
};

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] p) {
        int n = p.length,ans=1;
        Arrays.sort(p, (a,b)-> a[1]-b[1]);
        int[] prev = p[0];
        for(int i=1;i<n;i++){
            if (p[i][0] > prev[1]) {
                ans++;
                prev = p[i];
            }
        }
        return ans;
    }
}